cli: don't filter on program by default in merge-logs

tbg opened the pull request:
This behavior can be useful, but more often than not, the glob you're
passing in already matches exactly the right files. If the cockroach
binary wasn't called `cockroach` (this does happen), the previous
default wouldn't match any log files and gave a confusing UX.

Instead, accept any program by default. In cases where a filter is
adequate, one can be specified.

Release note: None

ajwerner reviewed (approved):
Fine by me. It may have minor conflicts with https://github.com/cockroachdb/cockroach/pull/33067 which attempts to deal with some of your other concerns. Also, I can update the above PR to return an error if no files match.

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/33319#-:-LUGEXoS0Mi1Uk_7HeAA:b1i4mtx)* status: :shipit: complete! 0 of 0 LGTMs obtained



<!-- Sent from Reviewable.io -->

tbg commented:
TFTR. I'll hold off on this until you've merged your other work. I do think
that the use case in which we're only looking at CRDB logs is the most
frequent one, so returning an error (while generally a good idea) wouldn't
solve the UX problem. The other default I was considering was
`^cockroach.*$`.

On Fri, Dec 21, 2018 at 3:51 PM ajwerner <notifications@github.com> wrote:

> *@ajwerner* approved this pull request.
>
> Fine by me. It may have minor conflicts with #33067
> <https://github.com/cockroachdb/cockroach/pull/33067> which attempts to
> deal with some of your other concerns. Also, I can update the above PR to
> return an error if no files match.
>
> *Reviewable
> <https://reviewable.io/reviews/cockroachdb/cockroach/33319#-:-LUGEXoS0Mi1Uk_7HeAA:b1i4mtx>*
> status: [image: :shipit:] complete! 0 of 0 LGTMs obtained
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/cockroachdb/cockroach/pull/33319#pullrequestreview-187420109>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AE135OyoP6W7QUph8u0tPTpZRA1ax7jkks5u7PVogaJpZM4ZdwNt>
> .
>

ajwerner reviewed (commented):
I support `^cockroach.*$`. I'll merge the other change now

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/33319#-:-LUGJrkD4xhimx6BzWN3:b70gx5n)* status: :shipit: complete! 0 of 0 LGTMs obtained



<!-- Sent from Reviewable.io -->

tbg committed:
cli: improve default program for merge-logs

The cockroach binary is often named `cockroach-something`, so this
should be picked up by the default regexp.

Release note: None

tbg commented:
Changed to `^cockroach.*$`. TFTR!

bors r=ajwerner
